File: test_ege_november/n24.cpp
#pragma GCC optimize("Ofast,unroll-loops")

#include <bits/stdc++.h>

#define ll long long
#define ld long double

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    ios::sync_with_stdio(false);

    ll left = 0, right = 11, sum = 0, len, counter = 0;
    string input;
    ifstream in;
    ofstream out;


    in.open("24.txt");
    in >> input;
    len = 1100000;
    vector<ll> m(1100000, 0);
    for (int i = 0; i < len; ++i) {
        if (input[i] == 'E') {
            m[i] = 1;
        } else if (input[i] == 'F') {
            m[i] = 9;
        } else {
            m[i] = 0;
        }
    }


    while (true) {
        if (sum == 2) {
            if (m[left] == 1 and m[right] == 1) {
                counter++;
                if (right - left > 11) {
                    sum -= m[left];
                    left++;
                } else {
                    if ()
                }
            }
        }
    }
}

File: test_ege_november/n24.py
with open("cmake-build-debug/24.txt") as f:
    a = f.read().strip()
s = []
for i in a:
    if i == 'E':
        s.append(1)
    elif i == 'F':
        s.append(9999)
    else:
        s.append(0)
a = ""
left = 0
right = 11
sum = 0
counter = 0
while True:
    if sum < 2:
        right += 1
        sum += s[right]
    elif sum > 2:
        if right - left > 12:
            sum -= s[left]
            left += 1
        else:
            if right < len(s) - 1:
                right += 1
                sum += s[right]
            else:
                break
    else:
        if s[left] == 1 and s[right] == 1:
            counter += 1
        if right < len(s) - 1:
            sum -= s[left]
            left += 1
            right += 1
            sum += s[right]
        else:
            break
print(counter)
